Source/Extensions/Blazorise.DataGrid/Models/DataGridRowInfo.cs

```diff
@@ -0,0 +1,29 @@
+﻿#region Using directives
+#endregion
+
+namespace Blazorise.DataGrid
+{
+    /// <summary>
+    /// Holds the basic information about the datagrid row.
+    /// </summary>
+    public class DataGridRowInfo<TItem>
+    {
+        private bool showDetail;
+        /// <summary>
+        /// Initializes a new instance of row info.
+        /// </summary>
+        /// <param name="item">Row Item</param>
+        public DataGridRowInfo( TItem item )
+        {
+            Item = item;
+        }
+
+        public TItem Item { get; }
+
+        public bool ShowDetail => showDetail;
+
+        public void SetShowDetail(bool toShowDetail)
```

Comment:
Rename to SetRowDetail or SetRowDetailVisibility. Also might be good to have ToggleRowDetail